src/ui/public/share/directives/share.js

```diff
@@ -144,9 +143,12 @@ app.directive('share', function (Private) {
         this.urlFlags.shortSnapshot = !this.urlFlags.shortSnapshot;
 
         if (this.urlFlags.shortSnapshot) {
-          urlShortener.shortenUrl(this.urls.snapshot)
+          shortenUrl(this.urls.snapshot)
```

Comment:
I kind of thought we decided to cast es6 promises to angular promises, rather than calling `$scope.$apply()`?

```js
Promise.resolve(shortenUrl(this.urls.snapshot))
  .then(shortUrl => {
    this.urls.shortSnapshot = shortUrl;
  })
```

Reply:
we could but it will not matter in the next couple of days since the next step for EUI and rectifying the share top nav is to rip this directive out replace with a react component. This is just temporary.

Reply:
Just for other efforts similar to this, if we can avoid using `$scope.$apply()` we should.